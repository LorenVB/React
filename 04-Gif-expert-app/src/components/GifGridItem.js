import React from 'react'
import PropTypes from 'prop-types'

export const GifGridItem = ( {title, url} ) => {
  
    return (
    <div>
       <img className="gif-img animate__animated animate__bounce animate__fadeIn" src={url} alt={ title } />
       <p> {title} </p>
    </div>
  )
}

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
}


