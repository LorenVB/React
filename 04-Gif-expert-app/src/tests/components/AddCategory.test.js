import React from 'react';
import {shallow} from 'enzyme'
import { AddCategory } from '../../components/AddCategory'

describe('Testing AddCategory Component', () => {

    const setCategories = jest.fn();
    let wrapper;

    beforeEach(() => {
         wrapper = shallow(<AddCategory setCategories={setCategories} />);
    })

    test('should render correctly', () => {
       
        expect(wrapper).toMatchSnapshot();
    })

    test('should not save the information after click the submit', () => {
        
        wrapper.find('form').simulate('submit', {preventDefault(){}});
        expect(setCategories).not.toHaveBeenCalled();
        
    })

    test('should call inputChange action and cleaning the textbox', () => {

        const value = "Hola Mundo";
        wrapper.find('input').simulate('change', {target: {value}});
        wrapper.find('form').simulate('submit', {preventDefault(){}});
        
        expect(setCategories).toHaveBeenCalled();
        expect(wrapper.find('input').prop('value')).toBe('');

    })

})