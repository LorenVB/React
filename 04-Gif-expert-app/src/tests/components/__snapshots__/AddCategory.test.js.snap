// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing AddCategory Component should render correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
